File: chazm-model/src/main/java/runtimemodels/chazm/model/entity/PolicyEntity.java
package runtimemodels.chazm.model.entity;

import com.google.inject.assistedinject.Assisted;
import runtimemodels.chazm.api.entity.Policy;
import runtimemodels.chazm.api.id.UniqueId;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;

class PolicyEntity extends AbstractEntity<Policy> implements Policy {

    @Inject
    PolicyEntity(@NotNull @Assisted final UniqueId<Policy> id) {
        super(id);
    }

    @Override
    public boolean equals(final Object object) {
        if (object instanceof Policy) {
            final Policy policy = (Policy) object;
            return super.equals(policy);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return super.toString();
    }

}

File: chazm-model/src/main/java/runtimemodels/chazm/model/entity/PolicyEntity.kt
package runtimemodels.chazm.model.entity

import com.google.inject.assistedinject.Assisted
import runtimemodels.chazm.api.entity.Policy
import runtimemodels.chazm.api.id.UniqueId

import javax.inject.Inject

internal open class PolicyEntity @Inject constructor(@Assisted id: UniqueId<Policy>) : AbstractEntity<Policy>(id), Policy {

    override fun equals(other: Any?): Boolean {
        if (other is Policy) {
            return super.equals(other)
        }
        return false
    }

    override fun hashCode(): Int = super.hashCode()

    override fun toString(): String = super.toString()

}
